phoenix-core/src/it/java/org/apache/phoenix/end2end/MetaDataEndPointIT.java

```diff
@@ -54,4 +60,107 @@ public void testUpdateIndexState() throws Throwable {
+            long expectedCacheUsedSize =
+                    IndexMetricsIT.getCounterValueByName(MetricsMetadataSource.METADATA_CACHE_USED_SIZE, registry);
+            PTable table = conn.getTableNoCache(fullTableName);
+
+            ddl = "DROP TABLE " + fullTableName;
+            conn.createStatement().execute(ddl);
+
+            expectedDropTableCount += 1;
+            IndexMetricsIT.verifyCounterWithValue(MetricsMetadataSource.DROP_TABLE_COUNT,
+                    registry, expectedDropTableCount);
+
+            expectedCacheUsedSize -= table.getEstimatedSize();
+            IndexMetricsIT.verifyCounterWithValue(MetricsMetadataSource.METADATA_CACHE_USED_SIZE,
+                    registry, expectedCacheUsedSize);
+        }
+    }
+
+    @Test
+    public void testMetadataMetricsOfAlterTableAddCol() throws Throwable {
+        String schemaName = generateUniqueName();
+        String tableName = generateUniqueName();
+        final String fullTableName = SchemaUtil.getTableName(schemaName, tableName);
+        try (PhoenixConnection conn = (PhoenixConnection) DriverManager.getConnection(getUrl())) {
+            MetricsMetadataSourceImpl metricsSource = (MetricsMetadataSourceImpl) MetricsMetadataSourceFactory.getMetadataMetricsSource();
+            DynamicMetricsRegistry registry = metricsSource.getMetricsRegistry();
+
+            String ddl = "CREATE TABLE " + fullTableName +
+                    "  (a_string varchar not null, col1 integer" +
+                    "  CONSTRAINT pk PRIMARY KEY (a_string)) ";
+            conn.createStatement().execute(ddl);
+
+            long expectedDropTableCount =
+                    IndexMetricsIT.getCounterValueByName(MetricsMetadataSource.ALTER_ADD_COLUMN_COUNT, registry);
+//            long expectedCacheUsedSize =
+//                    IndexMetricsIT.getCounterValueByName(MetricsMetadataSource.METADATA_CACHE_USED_SIZE, registry);
+//            int tableEstimatedSize = conn.getTableNoCache(fullTableName).getEstimatedSize();
+
+            ddl = "ALTER TABLE " + fullTableName + " ADD  b_string VARCHAR  NULL PRIMARY KEY  ";
+            conn.createStatement().execute(ddl);
+//            int newTableEstimatedSize = conn.getTableNoCache(fullTableName).getEstimatedSize();
+
+            expectedDropTableCount += 1;
+            IndexMetricsIT.verifyCounterWithValue(MetricsMetadataSource.ALTER_ADD_COLUMN_COUNT,
+                    registry, expectedDropTableCount);
+
+//            System.out.println(expectedCacheUsedSize);
+//            System.out.println(tableEstimatedSize);
+//            System.out.println(newTableEstimatedSize);
+//            expectedCacheUsedSize = expectedCacheUsedSize - tableEstimatedSize + newTableEstimatedSize;
+//            IndexMetricsIT.verifyCounterWithValue(MetricsMetadataSource.METADATA_CACHE_USED_SIZE,
+//                    registry, expectedCacheUsedSize);
```

Comment:
In this test, it’s hard to calculate `expectedCacheUsedSize` after altering table, because the size of newDeletedTableMarker(kv.getTimestamp()) is also included in the actual cache used size, and its value is hard to get, right? @virajjasani  WDYT?

Reply:
Correct @jinggou, it is not required that we compare the exact cache used size metric, but what we can do is measure that the value of used cache is higher or lower from previous value.